framework/src/main/java/org/checkerframework/common/value/ValueAnnotatedTypeFactory.java

```diff
@@ -1300,8 +1300,78 @@ private boolean methodIsStaticallyExecutable(Element method) {
             return getDeclAnnotation(method, StaticallyExecutable.class) != null;
         }
 
+        /**
+         * @return The Range of the Math.min or Math.max method, or null if the argument is none of
+         *     these methods or their arguments are not annotated with @IntRange.
+         */
+        private Range getRangeForMathMinMax(MethodInvocationTree tree) {
+            if (getMethodIdentifier().isMathMin(tree, processingEnv)) {
+                AnnotatedTypeMirror arg1 = getAnnotatedType(tree.getArguments().get(0));
+                AnnotatedTypeMirror arg2 = getAnnotatedType(tree.getArguments().get(1));
+                if (arg1.hasAnnotation(IntRange.class) && arg2.hasAnnotation(IntRange.class)) {
+                    Range rangeArg1 =
+                            new Range(getFromValueFromIntRange(arg1), getToValueFromIntRange(arg2));
+                    Range rangeArg2 =
+                            new Range(getFromValueFromIntRange(arg2), getToValueFromIntRange(arg2));
+                    return rangeArg1.min(rangeArg2);
+                }
+            } else if (getMethodIdentifier().isMathMax(tree, processingEnv)) {
+                AnnotatedTypeMirror arg1 = getAnnotatedType(tree.getArguments().get(0));
+                AnnotatedTypeMirror arg2 = getAnnotatedType(tree.getArguments().get(1));
+                if (arg1.hasAnnotation(IntRange.class) && arg2.hasAnnotation(IntRange.class)) {
+                    Range rangeArg1 =
+                            new Range(getFromValueFromIntRange(arg1), getToValueFromIntRange(arg2));
+                    Range rangeArg2 =
+                            new Range(getFromValueFromIntRange(arg2), getToValueFromIntRange(arg2));
+                    return rangeArg1.max(rangeArg2);
+                }
+            }
+            return null;
+        }
+
+        /**
+         * @return The constant value of the Math.min or Math.max method, or null if the argument is
+         *     none of these methods or their arguments are not constants.
+         */
+        private Long getMathMinMaxExactValue(MethodInvocationTree tree) {
```

Comment:
If you use `org.checkerframework.common.value.ValueAnnotatedTypeFactory#getRange` then this method won't be needed.